Sources/Widgets/Representations/SphereHandleRepresentation/index.js

```diff
@@ -49,12 +49,13 @@ function vtkSphereHandleRepresentation(publicAPI, model) {
   model.displayMapper.setInputConnection(publicAPI.getOutputPort());
   model.actors.push(model.displayActor);
   model.alwaysVisibleActors = [model.displayActor];
-
-  model.mapper = vtkGlyph3DMapper.newInstance({
-    scaleArray: 'scale',
-    colorByArrayName: 'color',
-    scalarMode: ScalarMode.USE_POINT_FIELD_DATA,
-  });
+  model.mapper = publicAPI.initializeMapper(
```

Comment:
`this initializeMapper` is not very userfriendly.

How about using `getActors()` or have a similar mechanism for mappers ?

I thought mappers could be initialized in `extend()`

Reply:
I think a similar mechanism for mappers might be redundant with `getActors`, as they are mostly initialized at the same place, but I'm not sure of exceptions ?

If we want to be able to use `getActors()`, then we need to initialize mappers and actors in `extend()`, and append them to the actor list in `extend()` anyways. This is because we want to automatically apply the topology parameters at the end of `WidgetRepresentation`'s `extend()`.

If that seems ok then let's do it like this.

Reply:
That implies making a rule that all mappers always have to be initialized in extend, and all actors have to be added to the list. I'm not sure that's ok, and if it could be enforced @finetjul ?

Alternatively, we could implement a method to replace `model.actors.push(actor)` with something custom that also initializes the mapper, but that looks a lot like `initializeMapper`

Reply:
yes, if what you have in mind is something like that:
```
extend() {
   ...
   getActors().forEach(actor -> initializeMapper(actor.getMapper()))
}
```

Reply:
You're right, it can also be done in a `addActor()` convenient method.

I feel it is better to initialize as late a possible in order to have as much context as possible.

Reply:
Yes, that's what I was thinking.

Do you think there's a way to enforce that all mappers/actors should be created in extend ?

Reply:
I believe it is implicitly already the case (@floryst ?). Maybe that should simply be documented.

Reply:
Right now they are mostly initialized in the (child) representation's constructor, but that's too late for us if we want to go  the `getActors()` route.